### ISSUE TITLE
Callback was already called

### ISSUE THREAD

**Issue by achingbrain** [ipfs/js-datastore-s3#12]
Got this in a long running process..

```javascript
/app/node_modules/aws-sdk/lib/request.js:31
            throw err;
            ^

Error: Callback was already called.
    at /app/node_modules/async/internal/onlyOnce.js:9:32
    at blockstore.put (/app/node_modules/ipfs-bitswap/src/index.js:160:7)
    at ManagedUpload.opts.s3.upload [as callback] (/app/node_modules/datastore-s3/src/index.js:104:7)
    at Response.finishSinglePart (/app/node_modules/aws-sdk/lib/s3/managed_upload.js:667:12)
    at Request.<anonymous> (/app/node_modules/aws-sdk/lib/request.js:364:18)
    at Request.callListeners (/app/node_modules/aws-sdk/lib/sequential_executor.js:106:20)
    at Request.emit (/app/node_modules/aws-sdk/lib/sequential_executor.js:78:10)
    at Request.emit (/app/node_modules/aws-sdk/lib/request.js:683:14)
    at Request.transition (/app/node_modules/aws-sdk/lib/request.js:22:10)
    at AcceptorStateMachine.runTo (/app/node_modules/aws-sdk/lib/state_machine.js:14:12)
    at /app/node_modules/aws-sdk/lib/state_machine.js:26:10
    at Request.<anonymous> (/app/node_modules/aws-sdk/lib/request.js:38:9)
    at Request.<anonymous> (/app/node_modules/aws-sdk/lib/request.js:685:12)
    at Request.callListeners (/app/node_modules/aws-sdk/lib/sequential_executor.js:116:18)
    at Request.emit (/app/node_modules/aws-sdk/lib/sequential_executor.js:78:10)
    at Request.emit (/app/node_modules/aws-sdk/lib/request.js:683:14)
    at Request.transition (/app/node_modules/aws-sdk/lib/request.js:22:10)
    at AcceptorStateMachine.runTo (/app/node_modules/aws-sdk/lib/state_machine.js:14:12)
    at /app/node_modules/aws-sdk/lib/state_machine.js:26:10
    at Request.<anonymous> (/app/node_modules/aws-sdk/lib/request.js:38:9)
    at Request.<anonymous> (/app/node_modules/aws-sdk/lib/request.js:685:12)
    at Request.callListeners (/app/node_modules/aws-sdk/lib/sequential_executor.js:116:18)
```

**Comment by jacobheun** [comment]
@achingbrain what version of the aws-sdk do you have? Looking through their issues and the stack trace it looks like it's an issue there. I can try to track it down there as they're not currently able to reproduce it.

**Comment by achingbrain** [comment]
I'm using the latest (at the time of writing) - `v2.343.0`

**Comment by wehmoen** [comment]
I get this every time I try to run the example. Does someone have a workaround?

**Comment by jacobheun** [comment]
We're likely going to need to wrap the callbacks with `once` to prevent this. Looking at the aws-sdk issues history it looks like there's a history of this getting fixed and re-broken.

**Comment by jacobheun** [comment]
Fix released in 0.2.1.
